File: py-recitations-solutions/variables/quadratic.py
import math as m
import cmath as cm

def quadraticformula(a: float, b: float, c: float)->list:
    # Create a function for the quadratic formula
    d=b**2-4*a*c
    
    if d >= 0:
        root1 = 0.5*(-b+m.sqrt(d))/a
        root2 = 0.5*(-b-m.sqrt(d))/a
    elif d < 0: 
        root1 = 0.5*(-b+cm.sqrt(d))/a
        root2 = 0.5*(-b-cm.sqrt(d))/a

    roots = [root1,root2]
    return roots

if __name__ == "__main__":

    print("Finding the roots of the quadratic polynomial ax^2 + bx + c")

    # Ask the user for inputs a, b, and c.
    a,b,c = map(float,input("Enter numerical inputs for a, b, and c (a<SPACE>b<SPACE>c<ENTER>): \n").split())

    # Locate the two roots using the quadratic formula.
    roots = quadraticformula(a,b,c)

    # Output the result to the user.
    if roots[0] != roots[1]:
        print(f"Roots of the quadratic {a}x^2 + {b}x + {c}: {roots}")
    else:
        print(f"The single real root of the quadratic {a}x^2 + {b}x + {c}: {roots[0]}")

File: cpp-recitations-solutions/variables/quadratic.cpp
#include <iostream>
#include <cmath>
#include <complex>

int main() {

  double a, b, c;
  double root1, root2;

  std::cout << "Finding the roots of a quadratic polynomial ax^2 + bx + c" << std::endl;

  // Ask the user to input coefficients a, b, c.
  std::cout << "Enter numerical input for a, b, and c (a<ENTER>b<ENTER>c<ENTER>): "<<'\n';
  std::cin >> a >> b >> c;

  double d = pow(b,2) - 4*a*c;

  // Find the two roots using the quadratic formula.
  if (d < 0){
  std::cout << "Determinant is negative, imaginary output" << '\n';
  std::complex <double> root (-0.5*b/a, 0.5*sqrt(-1*d)/a);
  // Output the result to the user.
  std::cout << "Roots of the quadratic " << a << "x^2 + "<< b << "x + " 
  << c << ": " << '\n' << root.real() << " +/- " << root.imag() << "j" <<'\n'; 
  }
  else if (d>0){
  std::cout << "Determinant is positive, two real outputs" << '\n';
  root1 =0.5*(-b + sqrt(d))/a;
  root2 =0.5*(-b-sqrt(d))/a;
  // Output the result to the user.
  std::cout << "Roots of the quadratic " << a << "x^2 + "<< b << "x + " << c << ": " << '\n' << root1 << ", "  << root2;
  }
  else {
  std::cout << "Determinant is zero, single real output" << '\n';
  root1 = 0.5*(-b)/a;
  // Output the result to the user.
  std::cout << "Root of the quadratic " << a << "x^2 + "<< b << "x + " << c << ": " << '\n' << root1 << '\n';        
  }

  return 0;
  
}
